cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -896,4 +909,22 @@ static void copyRuntimeToken(LocationFactory locationFactory, Arguments args, Fi
     localizeResources.put(Constants.Security.Authentication.RUNTIME_TOKEN_FILE,
                           new LocalizeResource(runtimeTokenFile.toURI(), false));
   }
+
+  private void generateRuntimeToken(File tempDir, Map<String, LocalizeResource> localizeResources)
```

Comment:
I don't understand why we need to generate this token file for all programs now. This would impact everything and doesn't seem correct.

I think we should keep using `new AuthenticationContextModules().getMasterModule()` and keep copying over the secret on namespace creation for now. The secret isn't something we normally expect to change, let's handle it in a separate jira.

Reply:
Done